packages/blocks/src/page-block/const/move-block-config.ts

```diff
@@ -0,0 +1,92 @@
+import type { BlockElement } from '@blocksuite/lit';
+
+import {
+  getBlockSelectionBySide,
+  getNextBlock,
+  getPrevBlock,
+  getTextSelection,
+  pathToBlock,
+} from '../../note-block/utils.js';
+
+interface MoveBlockConfig {
+  name: string;
+  hotkey: string;
+  action: (blockElement: BlockElement) => void;
+}
+
+export const moveBlockConfig: MoveBlockConfig[] = [
+  {
+    name: 'Move Up',
+    hotkey: 'Mod-Alt-ArrowUp',
+    action: blockElement => {
+      const textSelection = getTextSelection(blockElement);
+      if (textSelection) {
+        const block = pathToBlock(blockElement, textSelection.from.path);
+        if (!block) return;
+        const prevBlock = getPrevBlock(block, block => !!block.model.text);
+        if (!prevBlock) return;
+        const parent = blockElement.page.getParent(prevBlock.model);
+        if (!parent) return;
+        blockElement.page.moveBlocks(
+          [block.model],
+          parent,
+          prevBlock.model,
+          true
+        );
+        return;
+      }
+      const blockSelection = getBlockSelectionBySide(blockElement, true);
+      if (blockSelection) {
+        const block = pathToBlock(blockElement, blockSelection.path);
+        if (!block) return;
+        const prevBlock = getPrevBlock(block);
+        if (!prevBlock) return;
+        const parent = blockElement.page.getParent(prevBlock.model);
+        if (!parent || parent.flavour === 'affine:page') return;
+        blockElement.page.moveBlocks(
+          [block.model],
+          parent,
+          prevBlock.model,
+          true
+        );
```

Comment:
You should return `true` if the operation success to prevent other handlers for the same key map.

Reply:
OK, I've done.